packages/amplication-server/src/core/resource/resource.service.ts

```diff
@@ -373,6 +379,192 @@ export class ResourceService {
     return resource;
   }
 
+  async createResourceEntitiesFromExistingResource(
+    args: CreateResourceEntitiesArgs,
+    user: User
+  ): Promise<Resource> {
+    const { targetResourceId, entitiesToCopy } = args.data;
+    const resource = await this.prisma.resource.findUnique({
+      where: {
+        id: targetResourceId,
+      },
+      include: {
+        entities: {
+          where: {
+            deletedAt: null,
+          },
+        },
+      },
+    });
+
+    if (!resource) throw new AmplicationError("Target Resource doesn't exist");
+
+    entitiesToCopy.map((entity) => {
+      if (resource.entities.find((x) => x.name === entity.name)) {
+        throw new AmplicationError(
+          `Entity ${entity.name} is already exist in resourceId: ${resource.id}, process abort`
+        );
+      }
+    });
+
```

Comment:
Should we check also if an entity is related to another entity not part of the copy?

Reply:
@overbit  It happened in the field creations: 
<img width="525" alt="image" src="https://github.com/amplication/amplication/assets/97830649/a4fe74f6-3bb3-4103-96f7-9230beec3605">

In this case, we create a non-relation field based on the relatedEntity id type.